src/Application/Provider/ProductVariantPriceProvider.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\PriceHistoryPlugin\Application\Provider;
+
+use Sylius\Component\Core\Exception\MissingChannelConfigurationException;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelPricingInterface;
+
+final class ProductVariantPriceProvider implements ProductVariantPriceProviderInterface
```

Comment:
I wonder whether it would not be more consistent and easier, given the later incorporation of this code into Sylius, if we decorated the `ProductVariantPriceCalculator` instead of creating a new provider, but I do not have a strong opinion, we can do it later as the part of refactorization